### PR TITLE
fix JSON patch and add logging [K8SSAND-472]

### LINKED ISSUE #44: CRDs are broken in v0.3.1
This is a follow up to #42. The CRDs are still broken. The json patches in `crd/patches/cassdc_config_patch.json` need to be updated since the new version of the CRDs have different paths for the properties that are patched.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-472) by [Unito](https://www.unito.io)
┆Fix Versions: k8ssandra-1.2.0

### PR DESCRIPTION
**Pull request by jsanda**
Fixed #44

### CODE REVIEW

File: config/crd/bases/cassandra.k8ssandra.io_cassandrabackups.yaml [hunk 1]
```diff
@@ -4213,8 +4213,6 @@ spec:
               startTime:
                 format: date-time
                 type: string
-            required:
```
Comment (burmanm) [inline config/crd/bases/cassandra.k8ssandra.io_cassandrabackups.yaml:4216]: Was this change intended? Not in the changelist.

Comment (jsanda) [reply config/crd/bases/cassandra.k8ssandra.io_cassandrabackups.yaml:4216]: Yes it was intended. I added an explanation in the commit log. Here is what I said:

>    I have changed the CassdcTemplateSpec to a pointer. I think that makes more
    sense going forward for situations where we need to determine whether the
    property has been set. I manually tested a backup/restore operation to make
    sure that the change does not break anything.

**Review by burmanm — APPROVED** [review]
